controllers/sandbox_controller.go

```diff
@@ -90,6 +90,13 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	// Check if already marked as expired to avoid repeated operations, including cleanups
+	if sandboxMarkedExpired(sandbox) {
+		log.Info("Sandbox is already marked as expired")
+		// Note: the sandbox won't be deleted if shutdown policy is changed to delete after expiration.
+		return ctrl.Result{}, nil
```

Comment:
This early return prevents the sandbox from being deleted if the `ShutdownPolicy` is changed to `Delete` *after* expiration. While this aligns with the "once expired, always expired" philosophy, it effectively freezes the resource configuration. Please ensure this limitation is acceptable and consider if a check for policy changes is needed here.

Reply:
As documented in this comment here, this is intentional.